app/lib/actions/katello/repository/filtered_index_content.rb

```diff
@@ -23,10 +23,10 @@ def run
             unit_ids = search_units(repo)
             ::Katello::Deb.import_all(unit_ids, repo)
           elsif repo.yum?
-            unit_ids = search_units(repo)
-            if input[:content_type] == 'srpm'
-              ::Katello::Srpm.import_all(unit_ids, repo)
+            if input[:content_type] == ::Katello::Srpm::CONTENT_TYPE
+              ::Katello::Srpm.import_for_repository(repo)
```

Comment:
why did you decide to move out the unit_ids lookup?  This causes the repository to be fully indexed (for srpms)

Reply:
I can't recall why I did that, but I think it was because of some issues with the unit search and pulpcore.

Reply:
actually looking at this, it seems hard coded to go straight to pulp2.  I don't think this was every properly converted to handle rpms for pulp3 either.    I think we need to refactor this quite a bit.  The caller of this (FinishUpload) passes in a filter for pulp2 that fetches all the RPMs/Srpms uploaded in the past 5 minutes.  But this won't work with pulp3, in fact pulp3 provides the href for uploaded contents, so we shouldn't really need to do this anyways.  We should be able to take the provided hrefs from the upload and pass it into this (or into a pulp3 related action).   I'm a bit skeptical that rpm uploads work properly tbh!

Reply:
Right, as of right now, there is only the pulp2 path for unit_search. 

And yup, completely forgot to account for paging in the list operation!

Reply:
and this is still waiting on an update to try to pass in the unit href's, correct?

Reply:
Yes, if we can extract a valid content_unit_href/pulp_id, then we should be able to use `import_all`.

Reply:
The alternative is to refactor the unit search to have both pulp2/pulpcore solutions.